tests/test_factory.py

```diff
@@ -298,7 +298,7 @@ def test_poetry_with_non_default_secondary_source_legacy(
     assert poetry.pool.get_priority("PyPI") is Priority.DEFAULT
     assert poetry.pool.has_repository("foo")
     assert isinstance(poetry.pool.repository("foo"), LegacyRepository)
-    assert [repo.name for repo in poetry.pool.repositories] == ["PyPI", "foo"]
+    assert {repo.name for repo in poetry.pool.repositories} == {"PyPI", "foo"}
```

Comment:
What is the reason for this change? Since repositories are ordered by convention, we might as well check the correct order.

Reply:
We validate the order in `test_repository_pool.py`. My rationale here is that the (tests in) `factory` should not have to change if we decide to change the repository order in `RepositoryPool` as this is an implementation detail of the object that it creates. #7658 therefore introduces set comparisons in `test_factory` (6x) but failed to do this consistently (3 remaining cases of list comparisons) which I corrected here. Alternatively, we can replace the set comparisons by list comparisons. Please let me know which option you prefer.